spec/ci-spec.js

```diff
@@ -12,133 +6,115 @@ const express = require('express');
 const wrench = require('wrench');
 const CSON = require('season');
 const apm = require('../lib/apm-cli');
-const {
-  nodeVersion
-} = require('./config.json');
+const { nodeVersion } = JSON.parse(fs.readFileSync('./config.json'));
```

Comment:
```suggestion
const { nodeVersion } = require('./config.js');
```

Node is able to require JSON files without any extra effort on our part, so this one can stay as is.

Reply:
Yes, it is able to but
1. it seems overall bad practice to mix up actual module importing (that might not even use CommonJS in the future, hopefully) and config loading
2. JSON undergoes stricter validation than any JS code

This was an intentional change, as indicated in the opening comment: `one rather shouldn't require JSON files so reading it synchronously`.

Reply:
While I can see where you are coming from, you've also got to consider that the way finding the module/file is handled differently.

Currently all tests here are failing, because it can't find `./config.js`, in general `fs` will try to load a file from the path of where the script was called, in this case it's looking for `config.js` at the root of the repository, whereas using a `require` is able to find the file to load, relative to the path of the file requiring it.

So to get tests passing you'd either need to resolve the path to `config.js`, otherwise `require` it as it was done.

Up to you on the solution there, but just an fyi.

Reply:
First off, the file was named `config.json`, not `config.js` so I'm not sure how `config.js` came to be in the whole discussion.
Second, regarding the file resolution... I know that the path resolution is different but I couldn't find anything _importing this js file_, so I had to assume that it is invoked on its own. Now I read up on how Node deals with this, in the CommonJS world, one would probably use `__dirname` to be always accurate. Unfortunately, there doesn't seem to be a catch-all way to do this; it either works for CommonJS or ESM. For the time being, perhaps CommonJS will do. I still prefer the "read the file and parse it" approach over hiding behind a `require` that ties our hands much more to the old way of things.

Reply:
Yeah sorry I misspoke there (This is why I shouldn't code review at 3am lol).
But I do get not wanting to hide this logic and more strictly parse it with an actual JSON parser. You may potentially be able to inspect the `process` values to see if the current file directory is exposed. But yeah we could stick with the `require` for now to get it working, and ideally later on find a way to do this better

Reply:
I added a commit with the usual "append the filename to the __dirname variable" logic, already present in this script. Please check it out.